content/tutorials/mqtt-snake.textile

```diff
@@ -0,0 +1,532 @@
+    <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+  </a>
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-go-step7
+
+blang[python].
+  Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @python ./controller.py@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+  <a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+    <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+  </a>
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-python-step7
+
+h2. Step 8 - Play Snake!
+
+With the setup complete, you can now add in the code for the actual snake game. Replace the contents of @snake.html@ with <span lang="javascript">"our Snake HTML code":https://github.com/ably/tutorials/blob/mqtt-snake-js-step8/mqtt-snake/snake.html</span><span lang="go">"our Snake HTML code":https://github.com/ably/tutorials/blob/mqtt-snake-go-step8/mqtt-snake/snake.html</span><span lang="python">"our Snake HTML code":https://github.com/ably/tutorials/blob/mqtt-snake-python-step8/mqtt-snake/snake.html</span> code. Replace REPLACE_WITH_YOUR_API_KEY with your API Key. When you reload the @snake.html@ page you should see your basic Snake game, based off of "an external source":https://gist.github.com/atimb/1449250 code. Run the controller as before, and you can now play! Use the arrow keys to move, press the space key to pause/play, and try to grow the Snake as large as you can.
+
+<a href="/images/tutorials/mqtt-snake.gif" target="_blank">
+  <img src="/images/tutorials/mqtt-snake.gif" style="width: 100%" alt="Playing Snake!">
+</a>
+
+blang[javascript].
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/mqtt-snake-js.
+
+blang[go].
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/mqtt-snake-go.
+
+blang[python].
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/mqtt-snake-python.
+
+We recommend that you clone the repo locally:
+
+bc[sh]. git clone https://github.com/ably/tutorials.git
+
+Checkout to the tutorial branch:
+
+blang[javascript].
+  bc[sh]. git checkout mqtt-snake-js
+
+blang[go].
+  bc[sh]. git checkout mqtt-snake-go
+
+blang[python].
+  bc[sh]. git checkout mqtt-snake-python
+
+And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @snake.js@ and your controller.
```

Comment:
Perhaps mention at the beginning of the tutorial that they can directly check the demo without implementing the complete tutorial themselves, then redirect them to this section?